### ISSUE TITLE
Setting multiple host variables from one exec line

### ISSUE THREAD

**Issue by Fusl** [AsydSolutions/ASYD#110]
It would be cool if we can set multiple variables at the same time with only one exec line.

Currently I am doing something like this:

```
var var1 = exec: cat value1
var var2 = exec: cat value2
var var3 = exec: cat value3
```

value1, value2 and value3 are each 3 different text files with one line in it.

I would like to do something similar to

```
var var1, var2, var3 = exec: cat value1 value2 value3`
```

Since the 3 text files have each a new line on the end, the text will be printed as

```
content1\n
content2\n
content3\n
```

ASYD saves var1 with "content1", var2 with "content2" and var3 with "content3".

Please, would you be so kind and implement this?

Thanks a lot!

**Comment by Choms** [comment]
Well, actually this is not that easy because ASYD can't know up to what point you want to store in the variable (in this case would be newline but can differ)

You can use "var from json/xml = ..." and format the output of the command as json or xml, that way you can store as much variables as you want

**Comment by Fusl** [comment]
Using `var from json/xml = ...` is not an acceptable way since there are not many applications printing data in the json or xml format.

Current way of setting variables ...

var something1 = exec: cat file1.txt file2.txt

... ignores the second line anyway. file1.txt contains "this", file2.txt contains "testing". something1 end up being "this" but does not contain "thistesting", "this testing", "this\ntesting" or any similar.

Since we would NOT break any current deploy scripts with this, why not do it? It's easy to implement and the newline character is a common thing for this.

**Comment by Choms** [comment]
Hi, sorry I had no time for check into this until now.

I just verified your example actually sets a variable with the value "thistesting", it does not ignores the second line (it does ignores the newline character)

If I used the newline character for what you are proposing, if file1.txt contains "this\ntesting" and your line is...

`var var1, var2 = exec: cat file1.txt file2.txt`

then var1 would be set to "this", var2 to "testing" and file2.txt would get ignored.

Cheers!
